README.md

# Document builder for your fast work
#### I will continue to work on the package, like and enjoy in the near future



## Example

```dart
import 'package:document_builder/document_builder.dart';
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';

class AboutScreen extends StatelessWidget {
  const AboutScreen({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: SafeArea(
        child: SingleChildScrollView(
            child: Document(
              options: [
                TitleOptions(text: 'Правила бойцовского клуба'),
                ListOptions(indexList: '1', list: [
                  'правило Бойцовского клуба: не упоминать о Бойцовском клубе.',
                  'правило Бойцовского клуба: не упоминать нигде о Бойцовском клубе.',
                  'правило Бойцовского клуба: боец крикнул «стоп», выдохся, отключился — бой окончен.',
                  ': в бою участвуют лишь двое.'
                ])
              ],
              builders: [
                DocumentBuilder<TitleOptions>(builder: (context, options) => TitleDoc(options: options)),
                DocumentBuilder<ListOptions>(builder: (context, options) => ListDoc(options: options))
              ],
            )
        ),
      ),
    );
  }
}
```


### SubTitle use with help `SubTitleDoc` and `SubTitleOptions`
```dart
SubTitleOptions(
  text: 'SubTitle',
  style: TextStyle(
    fontSize: 50
  )
)
..
...
..
DocumentBuilder<SubTitleOptions>(builder: (context, options) => SubTitleDoc(options: options))
```

### Text use with help `TextDoc` and `TextOptions`
```dart
TextOptions(
  text: 'SubTitle',
  style: TextStyle(
    fontSize: 50
  )
)
..
...
..
DocumentBuilder<TextOptions>(builder: (context, options) => TextDoc(options: options))
```


### Table use with help `TableDoc` and `TableOptions`
```dart
TableOptions(
  table: [
    ['1', '2', '3', '4', '5'],
    ['1.1', '1.2', '1.3', '1.4', '1.5'],
    ['2.1', '2.2', '2.3', '2.4', '2.5'],
    ['3.1', '3.2', '3.3', '3.4', '3.5'],
    ['4.1', '4.2', '4.3', '4.4', '4.5'],
  ]
)
..
...
..
DocumentBuilder<TableOptions>(builder: (context, options) => TableDoc(options: options))
```


# Add your custom widget 
```dart

class CustomElementOptions extends ElementOptions {
  final Widget icon;
  final String? text;
  final TextStyle style;
  final EdgeInsets padding;

  CustomElementOptions({
    required this.icon,
    this.text,
    this.style = const TextStyle(
        fontSize: 17,
        fontWeight: FontWeight.w700
    ),
    this.padding = const EdgeInsets.symmetric(vertical: 20)
  });
}

class CustomElementWidget extends StatelessWidget {
  final CustomElementOptions options;

  const CustomElementWidget({super.key, required this.options});

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: options.padding,
      child: Row(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          options.icon,
          SizedBox(width: 10,),
          Expanded(
            child: Text(options.text ?? '',
              style: options.style,
            ),
          )
        ],
      ),
    );
  }
}
```

### And use this
```dart
Document(
  options: [
    CustomElementOptions(
        icon: Icon(Icons.account_circle, size: 30,),
        text: 'The user has the right to be a user'
    )
  ],
  builders: [
    DocumentBuilder<CustomElementOptions>(
        builder: (context, options) => CustomElementWidget(options: options)
    ),
  ]
)
```

CHANGELOG.md
## 0.0.1

* TODO: Describe initial release.

## 0.0.2

* Check how this work

## 0.0.3

* Practice

## 0.0.4

* Delete left directories

## 0.0.5

* Refactoring all logic `DocumentBuilder`


## 0.0.6

* Create Documentation

## 0.0.7

* Add normal documentation

example/lib/main.dart
import 'package:document_builder/document_builder.dart';
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(App());
}

class App extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: ExampleScreen(),
    );
  }
}

class ExampleScreen extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: SafeArea(
        child: SingleChildScrollView(
          child: Container(
              padding: EdgeInsets.all(20),
              child: Document(options: [
                TitleOptions(text: 'Title your document'),
                CustomElementOptions(
                    icon: Icon(
                      Icons.account_circle,
                      size: 30,
                    ),
                    text: 'The user has the right to be a user')
              ], builders: [
                DocumentBuilder<CustomElementOptions>(
                    builder: (context, options) =>
                        CustomElementWidget(options: options)),
                DocumentBuilder<TitleOptions>(
                    builder: (context, options) => TitleDoc(options: options))
              ])),
        ),
      ),
    );
  }
}

class CustomElementOptions extends ElementOptions {
  final Widget icon;
  final String? text;
  final TextStyle style;
  final EdgeInsets padding;

  CustomElementOptions(
      {required this.icon,
      this.text,
      this.style = const TextStyle(fontSize: 17, fontWeight: FontWeight.w700),
      this.padding = const EdgeInsets.symmetric(vertical: 20)});
}

class CustomElementWidget extends StatelessWidget {
  final CustomElementOptions options;

  const CustomElementWidget({super.key, required this.options});

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: options.padding,
      child: Row(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          options.icon,
          SizedBox(
            width: 10,
          ),
          Expanded(
            child: Text(
              options.text ?? '',
              style: options.style,
            ),
          )
        ],
      ),
    );
  }
}
